### DISCUSSION TITLE
Launch argument priority

### DISCUSSION THREAD

**Discussion by Speedlulu** [mitmproxy/mitmproxy#5143] [Ideas]
Hi,

I've noticed that options defined in `<confdir>/config.yml` take precedence over options supplied on the cli with `--set`, is that intentional ?

If not I'd argue that cli values should take precedence.

Thanks for your answers !

**Comment by mhils** [comment]
I absolutely agree, in fact this is already been tracked in https://github.com/mitmproxy/mitmproxy/issues/5069. :-)

**Reply by Speedlulu** [reply]
Nice !

I must have missed something I thought the change to implement this was as simple as [this](https://github.com/Speedlulu/mitmproxy/commit/bee3e69f72acc247258dd9e861dd6dc0c258b8eb).

**Reply by Speedlulu** [reply]
I indeed have missed something, doing it this way negates the possibility to change `confdir`, wouldn't it be possible to just set `confdir` if it exists and then do what I linked ? Or simply do [this](https://github.com/Speedlulu/mitmproxy/commit/24e7614050c5b94dd4a214f5ea6e69f62157d08e), it is not elegant but it would work and since it's only ran once it should be okay.
